utils/tzif/src/parse/tzif.rs

```diff
@@ -376,65 +376,40 @@ where
-where
-    Input: Stream<Token = u8>,
-    Input::Error: ParseError<Input::Token, Input::Range, Input::Position>,
-{
-    count_min_max(charcnt, charcnt, any())
-        .map(|bytes: Vec<u8>| String::from_utf8_lossy(&bytes).into_owned())
-}
-
 /// A series of bytes constituting an array of
 /// NUL-terminated (0x00) time zone designation strings. The total
 /// number of bytes is specified by the "charcnt" field in the
 /// header.
 ///
-/// Splits each designation into a vector of [`String`] where each string
-/// starts at an index defined by a local time type record and ends at a
-/// NUL-terminator (0x00)
+/// Splits the list of bytes by the NULL-terminator (0x00) character
+/// and puts each designation into a [`String`].
 ///
 /// > e.g.
 /// > ```text
 /// > "LMT\u{0}HMT\u{0}MMT\u{0}IST\u{0}+0630\u{0}"
 /// > ```
 ///
-/// Note that two designations MAY overlap if one is a suffix
-/// of the other. The character encoding of time zone designation
-/// strings is not specified.
+/// Note that a local time record index might point in the middle of a
+/// designation. In that case the record's designation is the specified
+/// suffix. The [DataBlock::time_zone_designation] method can be used to
+/// access the correct designation string given an index.
 ///
+/// The character encoding of time zone designation strings is not specified.
 /// However, time zone designations SHOULD consist of at least three (3) and no
 /// more than six (6) ASCII characters from the set of alphanumerics,
 /// '-', and '+'. This is for compatibility with POSIX requirements
 /// for time zone abbreviations, so this parser enforces a UTF-8 ASCII encoding,
 /// to ensure compatability with Rust strings.
-fn time_zone_designations<Input>(
-    charcnt: usize,
-    local_time_type_records: Vec<LocalTimeTypeRecord>,
-) -> impl Parser<Input, Output = Vec<String>>
+fn time_zone_designations<Input>(charcnt: usize) -> impl Parser<Input, Output = Vec<String>>
 where
     Input: Stream<Token = u8>,
     Input::Error: ParseError<Input::Token, Input::Range, Input::Position>,
 {
-    raw_time_zone_designations(charcnt).map(move |raw_time_zone_designations| {
-        let mut time_zone_designations = Vec::with_capacity(local_time_type_records.len());
-        for record in &local_time_type_records {
-            for end_idx in record.idx..charcnt {
-                if raw_time_zone_designations.as_bytes()[end_idx] == b'\0' {
-                    time_zone_designations.push(
-                        String::from_utf8_lossy(
-                            raw_time_zone_designations[record.idx..end_idx].as_bytes(),
-                        )
-                        .into_owned(),
-                    );
-                    break;
-                }
-            }
-        }
-        time_zone_designations
+    count_min_max(charcnt, charcnt, any()).map(|bytes: Vec<u8>| {
+        bytes
+            .split_inclusive(|b| *b == 0)
```

Comment:
> 🤏 nitpick: style

I would prefer to see `*b == b'\0'` here to preserve the relationship to the idea that this is a C-style null terminator character.